### PR TITLE
Volume quota refactor bz 1669555

### PR DESCRIPTION
**Pull request by patrickdillon**
Refactor #11131 

Simplify logic and improve readability by tarring volume- & node-configs then md5sum the archive, rather than checking both files separately. 

Fixes [bug 1669555](https://bugzilla.redhat.com/show_bug.cgi?id=1669555)

### CODE REVIEW

**Comment by sdodson** [comment]
What do you think about something like this for the comparison bits?

```
#!/bin/bash

restart_required=false
files_to_compare="node-config.yaml volume-config.yaml"

#pushd /etc/origin/node
for f in ${files_to_compare}; do
  # If either file exists, compare them, if non-zero return see if the
  # file exists in the extraction, copy into place if the file doesn't
  # exist in extraction then it was removed from the configmap, remove it
  if [[ -f "./${f}" || -f "./tmp/${f}" ]]; then
    c=$(cmp -s ./${f} ./tmp/${f})
    if [[ $? -ne 0 ]]; then
       restart_required=true
       if [ -f "./tmp/${f}" ]; then
         echo "copied ./tmp/${f} ./${f}"
         cp ./tmp/${f} ./${f}
       else
         echo "removing ./${f}"
         rm ./${f}
       fi
    fi
  fi
done
echo "restart_required=${restart_required}"
#popd
```

Quick test
```
$ echo A > node-config.yaml
$ echo A > tmp/node-config.yaml
$ ./test.sh 
restart_required=false
$ echo B > tmp/node-config.yaml
$ ./test.sh 
copied ./tmp/node-config.yaml ./node-config.yaml
restart_required=true
$ echo C > tmp/volume-config.yaml
$ ./test.sh 
copied ./tmp/volume-config.yaml ./volume-config.yaml
restart_required=true
$ rm tmp/volume-config.yaml 
$ ./test.sh 
removing ./volume-config.yaml
restart_required=true
```

**Comment by sdodson** [comment]
/lgtm cancel

Sorry for providing a conflicting review with @mtnbikenc. I think the originally proposed changes are an improvement but it still felt very hard to follow.

The configmap extraction probably doesn't remove files, so that's probably something we either need to consider cleaning up tmp/ after each run or just get rid of the file removal code I proposed. I doubt someone will create a volume-config.yaml and later remove it.

File: roles/openshift_node_group/files/sync.yaml [hunk 1]
```diff
@@ -124,35 +118,26 @@ spec:
...
-              cat /dev/null > /tmp/.old-volume-config
-            fi
-
-            md5sum /etc/origin/node/tmp/node-config.yaml > /tmp/.new
-
-            if [[ ! -f /etc/origin/node/tmp/volume-config.yaml ]]; then
-              cat /dev/null > /tmp/.new-volume-config
+            
+            tmp_path=/etc/origin/node/tmp
+            if [[ -f ${tmp_path}/volume-config.yaml ]]; then
```
Comment (vrutkovs) [inline roles/openshift_node_group/files/sync.yaml]: It would be best to tar node-config and use `tar --append` to append volume-config if it exists
